apps/govquests-frontend/src/domains/questing/components/QuestButton.tsx

```diff
@@ -0,0 +1,58 @@
+import Button from "@/components/ui/Button";
+import SignInButton from "@/domains/authentication/components/SignInButton";
+import { cn } from "@/lib/utils";
+import type React from "react";
+
+interface QuestButtonProps {
+  status: "unstarted" | "started" | "completed";
+  isSignedIn: boolean;
+  onClaim: () => void;
+}
+
+const QuestButton: React.FC<QuestButtonProps> = ({
+  status,
+  isSignedIn,
+  onClaim,
+}) => {
+  const getButtonConfig = () => {
+    switch (status) {
+      case "completed":
+        return {
+          text: "Claim Rewards",
+          action: onClaim,
+          disabled: false,
+        };
+      default:
+        return {
+          text: "Read the texts to claim",
+          action: () => {},
+          disabled: true,
+        };
+    }
+  };
+
+  const { text, action, disabled } = getButtonConfig();
```

Comment:
why use this pattern of calling a function instead of e.g. using useMemo?

Reply:
I didn't use the useMemo, because it's a very simple function, so I don't want to add complexity and memorization on it, but considering that this `getButtonConfig` will get more complexity with time, I think that make sense akredy uses useMemo here.